src/org/parosproxy/paros/extension/manualrequest/http/impl/HttpPanelSender.java

```diff
@@ -311,4 +308,12 @@ public URI getInvalidRedirection() {
             return invalidRedirection;
         }
     }
+    
+    public void setUseTrackingSessionState (boolean shouldUseTrackingSessionState) {
+        getButtonUseTrackingSessionState().setEnabled(shouldUseTrackingSessionState);
+        delegate.setUseGlobalState(shouldUseTrackingSessionState);

```

Comment:
- Needs a null check for `delegate`.
 - This should be done inside the following if statement (otherwise it could be set without the button being selected).
 - Per the talk in IRC, yeah, we also need to add an item listener to the button to enable/disable the global state in the sender (previously you had to close/open the dialogue to active it...).

Reply:
Should I just remove the condition and simplify to:
```Java
public void setUseTrackingSessionState (boolean shouldUseTrackingSessionState) {
    getButtonUseTrackingSessionState().setSelected(shouldUseTrackingSessionState);
    getDelegate().setUseGlobalState(shouldUseTrackingSessionState);
}
```
Whether it's false or not set it, and use getDeligate so that we don't have to worry about null handling.

Reply:
GAH, I think this should be using setSelected not setEnabled.

I can use getDelegate() to address the null handling. I think there's other stuff missing too...I'll have to tackle further changes next week I think.

Reply:
Fine with the use of `getDelegate` (though with null check it would avoid creating it). The current usage of setEnabled/setSelected is correct.

Reply:
Ok I figured if we were checking we’d want it to exist.

Reply:
Also I don’t think the setEnabled is right because if it starts off it can never be toggled....

Reply:
It can if the option is enabled (otherwise the user is not supposed to interact with the button).

Reply:
Oh I figured they should be able to toggle on or off at will totally separate from the global option, but to be honest it was a pretty shallow thought ;)